### PR TITLE
IGAPP-610: Weird language item spacing if some are not available

### PR DESCRIPTION
**Pull request by maxammann**
… text when hovering

This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by maxammann** [comment]
This PR also fixes an issue with the hover effect. If you hover an element it becomes bold and increases the width. The BoldSpacer fixes this jumping.

**Comment by maxammann** [comment]
I also increased the max width of tooltip content

**Review by SteffiStuffel — APPROVED** [review]

**Comment by SteffiStuffel** [comment]
Tested on Chrome

File: web/src/modules/common/components/Selector.js [hunk 1]
```diff
@@ -30,22 +30,25 @@ const Element: StyledComponent<{||}, ThemeType, *> = styled(Link)`
     font-size: 1em;
     line-height: ${dimensions.headerHeightSmall}px;
   }
-`
+  
+  ${props =>
+    props.enabled ? `color: ${props.theme.colors.textColor};` : `color: ${props.theme.colors.textDisabledColor};`}
```
Comment (klinzo) [inline web/src/modules/common/components/Selector.js:35]:
```suggestion
  color: ${props => props.enabled ? props.theme.colors.textColor : props.theme.colors.textDisabledColor};
```

File: web/src/modules/common/components/Selector.js [hunk 2]
```diff
@@ -30,22 +30,25 @@ const Element: StyledComponent<{||}, ThemeType, *> = styled(Link)`
...
-const ActiveElement: StyledComponent<{| selected: boolean |}, ThemeType, *> = styled(Element)`
-  color: ${props => props.theme.colors.textColor};
   ${props =>
     props.selected
       ? 'font-weight: 700;'
       : `:hover {
           font-weight: 700;
           border-radius: 0;
         }`}
+}
```
Comment (klinzo) [inline web/src/modules/common/components/Selector.js:44]: I think this bracket is to much:
```suggestion
```

File: web/src/modules/common/components/Tooltip.js [hunk 3]
```diff
@@ -152,13 +152,16 @@ const TooltipContainer: StyledComponent<
...
+type PropsType = {
   children: React.Node,
   text: ?string,
   flow: FlowType,
   mediumViewportFlow?: FlowType,
-  smallViewportFlow?: FlowType
-|}
+  smallViewportFlow?: FlowType,
+  className?: string
+}
```
Comment (klinzo) [inline web/src/modules/common/components/Tooltip.js:164]:
```suggestion
  className?: string,
  ...
}
```

File: web/src/modules/common/components/Tooltip.js [hunk 4]
```diff
@@ -152,13 +152,16 @@ const TooltipContainer: StyledComponent<
   }
 `
 
-type PropsType = {|
+// We do not want exact types here as we forward props
+// eslint-disable-next-line flowtype/require-exact-type
```
Comment (klinzo) [inline web/src/modules/common/components/Tooltip.js:156]:
```suggestion
```

**Review by klinzo — APPROVED** [review]
I think you could add a release note for this :)

One minor observation: The tooltip for chinese (http://0.0.0.0:9000/testumgebung/de/news/local/157) was hidden behind the langauage change icon for me once, but I was not able to reproduce it again :/ So I think you can just ignore it.

**Review by Taggotty — CHANGES_REQUESTED** [review]
I get the following error in the web console: `Warning: Received 'true' for a non-boolean attribute 'enabled'.`

**Comment by klinzo** [comment]
> I get the following error in the web console: `Warning: Received 'true' for a non-boolean attribute 'enabled'.`

I think we had this already, the reason are styled component props if I remember corectly: https://styled-components.com/docs/api#transient-props

**Comment by maxammann** [comment]
@klinzo I imporoved the positioning of tooltips

**Review by klinzo — APPROVED** [review]
Tested, looks really good!

**Review by Taggotty — APPROVED** [review]
For the language tooltip I still get an overflow on the right side. Personally I think that is no big problem, but if you find a quick fix I am also content :)

**Comment by maxammann** [comment]
> For the language tooltip I still get an overflow on the right side. Personally I think that is no big problem, but if you find a quick fix I am also content :)


Fixed, missed to remove a `/2`

**Pull request merged**